### ISSUE TITLE
Printing to statusline causes... problems

### ISSUE THREAD

**Issue by Xandaros** [LiquidityC/Blightmud#76]
![](http://tcp.st/kokv9.png)

I'm printing to the status line in response to a GMCP message (Char.Vitals). I reset the colour at the end of the string, just by virtue of how it is generated. The bar does not change its colour back to what it should be after the text - I would expect it to.

After printing to the line, the end of the bar is missing.
The cursor appears to be at the end of the visible bar, not on the input line. Typing anything moves it back down, though.

Complete relevant script:
```lua
local function formatStat(value, max)
    local color
    value = tonumber(value)
    max = tonumber(max)
    if value < 0.3 * max then
        color = C_BRED
    elseif value < 0.7 * max then
        color = C_BYELLOW
    else
        color = C_GREEN
    end
    return color .. value .. "/" .. max .. C_RESET
end

local function updateStatus()
    local vitals = GMCP.char.vitals
    local hp = formatStat(vitals.hp, vitals.maxhp)
    local mp = formatStat(vitals.mp, vitals.maxmp)
    local ep = formatStat(vitals.ep, vitals.maxep)
    local wp = formatStat(vitals.wp, vitals.maxwp)
    local line = hp .. " " .. mp .. " " .. ep .. " " .. wp

    blight:status_line(0, line)
end

blight:on_gmcp_ready(function()
    blight:output("GMCP enabled")
    blight:register_gmcp("Char")
    blight:add_gmcp_receiver("Char.Vitals", function(data)
        obj = json.decode(data)
        GMCP = GMCP or {}
        GMCP.char = GMCP.char or {}
        GMCP.char.vitals = obj
        updateStatus()
    end)
end)
```

**Comment by LiquidityC** [comment]
Yes that’s because you reset the color and it doesn’t handle that. It definitely should handle that so this is a definite bug. I’ll get on it during the week.

**Comment by LiquidityC** [comment]
I just realized you had a PR here. I’ll check that out as soon as I’m back from travels. 😉

**Comment by Xandaros** [comment]
Yeah, this was annoying me enough that Blightmud was essentially unusable for me. The colour is one thing, what really annoyed me was the cursor being in the wrong place. And I had that happen even without any colour codes in the status line, the cursor would end up on the last character of the bottom-most status line.

I figured this should be easy enough to fix myself, so I did. Turned out to be more complicated than expected, but that's how software works sometimes, isn't it :D
